server/protocols/jmap/src/main/java/org/apache/james/jmap/utils/MailboxHierarchySorter.java

```diff
@@ -19,35 +19,44 @@
 
 package org.apache.james.jmap.utils;
 
+import java.util.Collection;
 import java.util.List;
 import java.util.Map;
+import java.util.Optional;
 import java.util.function.Function;
 import java.util.stream.Collectors;
 
-import org.apache.james.jmap.model.mailbox.Mailbox;
-
 import com.google.common.collect.Lists;
 
-public class MailboxHierarchySorter {
+public class MailboxHierarchySorter<T, Id> {
```

Comment:
Agreed. The naming issue is  hiding the fact that the previously specialized class is no longer specialized as it is now using generics, and bringing little value compared to the underling class DependencyGraph.
A proper renaming is at least required IMO.